nullaway/src/main/java/com/uber/nullaway/fixserialization/XMLUtil.java

```diff
@@ -0,0 +1,202 @@
+      Document doc, String key, String attr, Class<T> klass) {
+    Node node = getNodeFromDocument(doc, key);
+    if (node == null) {
+      return new DefaultXMLValueProvider<>(null, klass);
+    }
+    return new DefaultXMLValueProvider<>(node.getAttributes().getNamedItem(attr), klass);
+  }
+
+  /**
+   * Helper method for reading value of a node located at key_1:key_2:...:key_n from a {@link
+   * Document}.
+   *
+   * @param doc XML object to read values from.
+   * @param key Key to locate the value, can be nested (e.g. key1:key2:...:key_n).
+   * @param klass Class type of the value in doc.
+   * @return The value in the specified keychain cast to the class type given in parameter.
+   */
+  public static <T> DefaultXMLValueProvider<T> getValueFromTag(
+      Document doc, String key, Class<T> klass) {
+    Node node = getNodeFromDocument(doc, key);
+    if (node == null) {
+      return new DefaultXMLValueProvider<>(null, klass);
+    }
+    return new DefaultXMLValueProvider<>(node, klass);
+  }
+
+  /**
+   * Writes the {@link FixSerializationConfig} in {@code XML} format.
+   *
+   * @param config Config file to write.
+   * @param path Path to write the config at.
+   */
+  public static void writeInXMLFormat(FixSerializationConfig config, String path) {
+    DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+    try {
+      DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+      Document doc = docBuilder.newDocument();
+
+      // Root
+      Element rootElement = doc.createElement("serialization");
+      doc.appendChild(rootElement);
+
+      // Suggest
+      Element suggestElement = doc.createElement("suggest");
+      suggestElement.setAttribute("active", String.valueOf(config.suggestEnabled));
+      suggestElement.setAttribute("enclosing", String.valueOf(config.suggestEnclosing));
+      rootElement.appendChild(suggestElement);
+
+      // Annotations
+      Element annots = doc.createElement("annotation");
+      Element nonnull = doc.createElement("nonnull");
+      nonnull.setTextContent(config.annotationConfig.getNonNull().fullName);
+      Element nullable = doc.createElement("nullable");
+      nullable.setTextContent(config.annotationConfig.getNullable().fullName);
+      annots.appendChild(nullable);
+      annots.appendChild(nonnull);
+      rootElement.appendChild(annots);
+
+      // Output dir
+      Element outputDir = doc.createElement("path");
+      outputDir.setTextContent(config.outputDirectory);
+      rootElement.appendChild(outputDir);
+
+      // Writings
+      TransformerFactory transformerFactory = TransformerFactory.newInstance();
+      Transformer transformer = transformerFactory.newTransformer();
+      DOMSource source = new DOMSource(doc);
+      StreamResult result = new StreamResult(new File(path));
+      transformer.transform(source, result);
+    } catch (Exception e) {
+      throw new RuntimeException("Error happened in writing config." + e);
```

Comment:
Same as before: a) Is `Exception` the most precise type possible? , b) Let's pass `e` as a cause.

Reply:
415a7dfe7c03a9fc2bd0f8ee1efe8618e9f6537e